driver/gamepad.c

```diff
@@ -322,16 +325,10 @@ static int gip_gamepad_op_firmware(struct gip_client *client, void *data, u32 le
     struct gip_gamepad *gamepad = dev_get_drvdata(&client->dev);
     struct input_dev *dev = gamepad->input.dev;
 
-    // mimic xpad behavior of ignoring if a profile is set
```

Comment:
To be clear, these profile checks were very intentionally removed.  xpad's behavior here is severely braindead - it ignores the fact that there are a wide variety of reasons a profile might be set that have nothing to do with paddle mapping:

- LED color/brightness customization
- Stick responsiveness customization
- Trigger responsiveness customization

Any of the above disable paddle support if they have been used, even if paddle mapping is not.